[Security] tests for API keys app

alisonelizabeth opened the pull request:
This PR adds tests for the new API keys app added via https://github.com/elastic/kibana/pull/45740.

Fixes https://github.com/elastic/kibana/issues/47869

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/5385/)
* Commit: 37d684ebd2aa72fa4dc93c442a8241c8eb86a405

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/5422/)
* Commit: 620c839412107bc703b539c91cb535b96347d896

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/5882/)
* Commit: 8d4347ef2528dfe2ec0d0efd6da93019e4674af8

alisonelizabeth commented:
@elasticmachine merge upstream

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/5909/)
* Commit: abd167b43b09f7d39b2d98137c567a89f0398bb3

alisonelizabeth commented:
@elasticmachine merge upstream

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/6580/)
* Commit: 93feb2c0c38b64c022ae01143b64cbc7df1f323c

alisonelizabeth commented:
@elasticmachine merge upstream

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/7303/)
* Commit: 059d95195bdd828867f406a4604a871513053be9

cuff-links commented:
Testing in progress.

alisonelizabeth commented:
@elasticmachine merge upstream

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/7998/)
* Commit: a2513eff6383bce855d365e745c165075d8043b3

kobelb commented on x-pack/legacy/plugins/security/public/views/management/api_keys_grid/components/api_keys_grid_page.test.tsx:
```diff
@@ -0,0 +1,182 @@
+  it('renders a loading state when fetching API keys', async () => {
+    const wrapper = mountWithIntl(<ApiKeysGridPage />);
+
+    expect(wrapper.find(SectionLoading)).toHaveLength(1);
```
Would using a `data-test-subj` here let us get away from having to use `// eslint-disable-next-line @kbn/eslint/no-restricted-paths` above?

kobelb commented:
Thanks so much for adding these @alisonelizabeth!

alisonelizabeth commented:
@elasticmachine merge upstream

alisonelizabeth commented:
@kobelb thanks for the review! I finally got around to addressing your feedback. I converted the API tests to jest and removed sinon. Do you think you could take another look?

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10472/)
* Commit: 0a9e3d5f01033ddf8149821bf92ab6b90a935981

alisonelizabeth commented:
@elasticmachine merge upstream

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/11050/)
* Commit: 919215565f1d17feefa09a03e927f97fc8dc3ca6
